pyro/infer/mcmc/hmc.py

```diff
@@ -185,11 +185,13 @@ def setup(self, *args, **kwargs):
         if self._automatic_transform_enabled:
             self.transforms = {}
         for name, node in sorted(trace.iter_stochastic_nodes(), key=lambda x: x[0]):
-            r_loc = torch.zeros_like(node["value"])
-            r_scale = torch.ones_like(node["value"])
-            self._r_dist[name] = dist.Normal(loc=r_loc, scale=r_scale)
+            site_value = node["value"]
             if node["fn"].support is not constraints.real and self._automatic_transform_enabled:
                 self.transforms[name] = biject_to(node["fn"].support).inv
+                site_value = self.transforms[name](node["value"])
+            r_loc = torch.zeros_like(site_value)
+            r_scale = torch.ones_like(site_value)
```

Comment:
Just wondering if `new_zeroes`, `new_ones` are better to correct device for these tensors?

Reply:
Good point. Will change.